pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -246,62 +353,18 @@ func (e *DockercfgController) syncServiceAccount(key string) error {
 		}
 
 		updatedSA, err := e.client.ServiceAccounts(serviceAccount.Namespace).Update(serviceAccount)
-		if err == nil {
-			e.serviceAccountCache.Mutation(updatedSA)
-		}
-		return err
-	}
-
-	dockercfgSecret, err := e.createDockerPullSecret(serviceAccount)
```

Comment:
I would expect the only change to `syncServiceAccount` to be to short-circuit if `createDockerPullSecret` says is didn't create the secret:

```
dockercfgSecret, created, err := e.createDockerPullSecret(serviceAccount)
if err != nil {
  return err
}
if !created  {
  return nil
}
```